src/oid.rs

```diff
@@ -134,8 +136,12 @@ impl ObjectId {
     // Generates a new timestamp representing the current seconds since epoch.
     // Represented in Big Endian.
     fn gen_timestamp() -> [u8; 4] {
-        let timespec = time::get_time();
-        let timestamp = timespec.sec as u32;
+        let timestamp = SystemTime::now()
+            .duration_since(SystemTime::UNIX_EPOCH)
+            .expect("system clock is before 1970")
+            .as_secs()
+            .try_into()
+            .expect("FIXME before 2038");
```

Comment:
since this buffer is actually an unsigned 32 bit field, this should succeed until aboout Jan 6th, 2106 according to the [ObjectId spec](https://github.com/mongodb/specifications/blob/master/source/objectid.rst#timestamp-field). I think we can update this "FIXME" to say something like "will succeed until 2106 due to unsigned" or something similar to reflect the low risk nature of the unwrap here.